tools/app/app/components/modals/DeleteModal.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useCallback, useEffect, useMemo } from 'react'
+import {
+  Checkbox,
+  Modal,
+  ModalDialog,
+  Typography,
+  DialogTitle,
+  DialogContent,
+  Divider,
+  DialogActions,
+  Button,
+  Alert,
+} from '@mui/joy'
+import { Trans, useTranslation } from 'react-i18next'
+import { Warning } from '@mui/icons-material'
+import { useCard } from '../../lib/api'
+import {
+  useAppDispatch,
+  useChildAmount,
+  useIsMounted,
+  useParentCard,
+} from '@/app/lib/hooks'
+import { useRouter } from 'next/navigation'
+import { addNotification } from '@/app/lib/slices/notifications'
+
+export interface DeleteModalProps {
+  open: boolean
+  onClose: () => void
+  cardKey: string
+}
+
+export function DeleteModal({ open, onClose, cardKey }: DeleteModalProps) {
+  const { t } = useTranslation()
+  const [checked, setChecked] = React.useState(false)
+
+  const { deleteCard } = useCard(cardKey)
+  const childAmount = useChildAmount(cardKey)
+
+  const dispatch = useAppDispatch()
+
+  const parent = useParentCard(cardKey)
+
+  const router = useRouter()
+
+  const isMounted = useIsMounted()
+
+  const warning = useMemo(
+    () =>
+      childAmount > 1
+        ? t('deleteCardModal.warning', { cardAmount: childAmount })
+        : undefined,
+    [childAmount, t]
+  )
+
+  const content = useMemo(
+    () => (
+      <Trans
+        i18nKey="deleteCardModal.content"
+        values={{
+          card: cardKey,
+        }}
+        count={childAmount}
+        components={{
+          bold: <Typography fontWeight="bold" />,
+        }}
+      />
+    ),
+    [cardKey, childAmount]
+  )
+
+  const handleDelete = useCallback(async () => {
+    try {
+      await deleteCard()
+      dispatch(
+        addNotification({
+          message: t('deleteCardModal.success', { card: cardKey }),
+          type: 'success',
+        })
+      )
+      if (isMounted()) {
```

Comment:
Why do you need to check if component is mounted?

Reply:
deleteCard-function might take a long time and onClose could be unsafe to call if the component is unmounted during that time. Might happen if connection is slow and user navigates before update is through.

Reply:
OK, got it! 👍